pkg/entities/tokens.go

```diff
@@ -2,92 +2,98 @@ package entities
+	"github.com/defipod/mochi/pkg/response"
 )
 
-type Response struct {
-	Id     string `json:"id"`
-	Symbol string `json:"symbol"`
-	Name   string `json:"name"`
-}
+func (e *Entity) CreateCustomToken(req request.UpsertCustomTokenConfigRequest) error {
+	token, _ := e.repo.Token.GetOneBySymbol(req.Symbol)
+	if token == nil {
+		chain, err := e.repo.Chain.GetByCurrency(req.Chain)
+		if err != nil {
+			return err
+		}
 
-func (e *Entity) GetIDAndName(symbol string) (string, string, error) {
-	resp, err := http.Get("https://api.coingecko.com/api/v3/coins/list")
-	// Get request
-	if err != nil {
-		return "", "", err
-	}
-	defer resp.Body.Close()
+		coinGeko, err := getCoinGeckoBySymbol(req.Symbol)
+		if err != nil {
+			return err
+		}
+		if coinGeko == nil {
+			return errors.New("cannot find token by symbol")
+		}
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", "", err
-	}
-	var res []Response
-	err1 := json.Unmarshal([]byte(body), &res)
-	if err1 != nil {
-		return "", "", err
-	}
-	for i := 0; i < len(res); i++ {
-		if res[i].Symbol == symbol {
-			return res[i].Id, res[i].Name, nil
+		token = &model.Token{
+			ChainID:             chain.ID,
+			Address:             req.Address,
+			CoinGeckoID:         coinGeko.ID,
+			Name:                coinGeko.Name,
+			Symbol:              strings.ToUpper(coinGeko.Symbol),
+			Decimals:            req.Decimals,
+			DiscordBotSupported: true,
+			Chain:               chain,
+		}
+		if err = e.repo.Token.CreateOne(token); err != nil {
+			return err
 		}
 	}
 
-	return "", "", err
-}
-
-func (e *Entity) CheckExistToken(symbol string) (bool, error) {
-	listSymbol, err := e.repo.Token.GetAll()
-	if err != nil {
-		return false, err
+	if !token.DiscordBotSupported {
+		return errors.New("token is not supported to add to your server")
 	}
 
-	for i, _ := range listSymbol {
-		if strings.ToLower(symbol) == strings.ToLower(listSymbol[i].Symbol) {
-			return true, nil
-		}
+	fmt.Println(token.Chain.Currency, req.Chain)
+	if !strings.EqualFold(token.Chain.Currency, req.Chain) {
+		return errors.New("token is not supported on this chain")
 	}
 
-	return false, nil
+	gct := model.GuildConfigToken{
+		GuildID: req.GuildID,
+		TokenID: token.ID,
+		Active:  true,
+	}
+	return e.repo.GuildConfigToken.CreateOne(gct)
 }
 
-func (e *Entity) CreateCustomToken(req request.UpsertCustomTokenConfigRequest) error {
-	err := e.repo.Token.CreateOne(model.Token{
-		Address:             req.Address,
-		Symbol:              strings.ToUpper(req.Symbol),
-		ChainID:             req.ChainID,
-		Decimals:            req.Decimals,
-		DiscordBotSupported: req.DiscordBotSupported,
-		CoinGeckoID:         req.CoinGeckoID,
-		Name:                strings.ToUpper(req.Name),
-		GuildDefault:        req.GuildDefault,
-	})
+func (e *Entity) GetAllSupportedToken(guildID string) (returnToken []model.Token, err error) {
+	returnToken, err = e.repo.Token.GetAllSupportedToken(guildID)
```

Comment:
log error everywhere

Reply:
this is old code, I don't change anything here